uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -2654,6 +2693,11 @@ public function testLeaveRequestIsValidShouldNotReturnErrorWhenValidationsPass()
       'end_date'   => CRM_Utils_Date::processDate('2016-12-31'),
     ]);
 
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contactID],
+      ['period_start_date' => '2016-01-01']
+    );
+
```

Comment:
Wow, this class is very long. I think we should impose some sort of limit in the future sometime. It could be split into separate test classes.